### PR TITLE
allow tf.Example to be created directly in t2t problem for tfidf

### PR DESCRIPTION
**Pull request by mark-radai**

### CODE REVIEW

File: tensor2tensor/data_generators/generator_utils.py [hunk 1]
```diff
@@ -46,6 +46,8 @@ def to_example(dictionary):
   for (k, v) in six.iteritems(dictionary):
     if not v:
       raise ValueError("Empty generated field: %s" % str((k, v)))
+    if str(type(v)) == "<class 'tensorflow.core.example.example_pb2.Example'>":
```
Comment (epurdy) [inline tensor2tensor/data_generators/generator_utils.py]: Better to do an isinstance() check, probably...

Comment (epurdy) [reply tensor2tensor/data_generators/generator_utils.py]: I'm pretty confused by how this is meant to work. Would it make more sense for `dictionary` to receive this treatment?

Comment (mark-radai) [reply tensor2tensor/data_generators/generator_utils.py]: yeah, valid.  I was being lazy about it. changed.  

~~yeah so the problem is that theres a disconnect inherent to tensor2tensor that doesnt allow a t2t problem's datagen to construct tf.train.Example s that describe sparse tensors.  as seen in to_example, it requires each feature to~~ 

in retrospect, yeah, there's def something wrong with this implementation.  hang on, I need to think a little on how to fix this.

the basic problem is that afaik tensor2tensor doesnt have built-in functionality for handling a sparse feature.  or rather it sort of does in the context of text tokens, but that's built under the (in this case) heavy restriction that theres only a single token passed per time step, which in a text2text problem is the index of the token.  It's possible to side-step this by just representing the tfidf vector densely, but....that pretty quickly and violently murders performance, especially if one wants to consider anything higher than 1-grams

Comment (mark-radai) [reply tensor2tensor/data_generators/generator_utils.py]: possibly addressed this issue by specifying `sample` keys to be used for the args in `data_items_to_decoders = {'targets': tf.contrib.slim.data_decoders.SparseTensor(<args>)}` in the tfidf problem's `example_reading_spec()` method.  I had originally thought that the SparseTensor decoder needed to operate on a separate `tf.train.Example` in order to function, but according to examples I found in tensor2tensor, that is not the case.

**Pull request closed without merging**